### PR TITLE
gdb: fix printing binary strings encoded in MsgPack

### PR DESCRIPTION
**Pull request by locker**
One may encode a non-printable string under the MsgPack type MP_STR.
In fact, there's no way to encode binary data as MP_BIN from Lua so
users often do that, which breaks our gdb extension.

Let's print a string as base64 data if unicode() raises an error.
While we are at it, let's also escape double quotes in printable
strings.

Here's what it looks like:

```
tarantool> box.tuple.new{"\"тест\\foo\"",
         >               require('msgpack').encode({})}
         ---
         - ['"тест\foo"', !!binary kA==]
         ...

(gdb) p *box_tuple_last
$1 = {local_refs = 3 '\003', flags = 0x00, format_id = 0,
  <...>
  data = ["\"тест\\foo\"", !!binary kA==]}
```

Follow-up commit 08a171a4b5bd

### CODE REVIEW

**Review by elhimov — CHANGES_REQUESTED** [review]

File: tools/tarantool-gdb.py [hunk 1]
```diff
@@ -395,16 +395,23 @@ def to_string_data(cls, data, depth):
...
+            val = bytes(data.read(len))
+            is_bin = True if mp_type == cls.MP_BIN else False
+            if not is_bin:
+                try:
+                    val = unicode(val, 'utf-8')
+                except UnicodeDecodeError:
+                    is_bin = True
+
+            if is_bin:
+                s += '!!binary '
```
[the remark this answers is not in the document]
Comment (locker) [reply tools/tarantool-gdb.py]: Makes sense. Fixed.

File: tools/tarantool-gdb.py [hunk 2]
```diff
@@ -395,16 +395,23 @@ def to_string_data(cls, data, depth):
         elif mp_type == cls.MP_INT:
             s += str(cls.decode_int(data))
 
-        elif mp_type == cls.MP_STR:
+        elif mp_type == cls.MP_STR or mp_type == cls.MP_BIN:
             len = cls.decode_strl(data)
```
[the remark this answers is not in the document]
Comment (locker) [reply tools/tarantool-gdb.py:399]: https://github.com/tarantool/tarantool/pull/8340#discussion_r1111679527

File: tools/tarantool-gdb.py [hunk 3]
```diff
@@ -397,14 +397,28 @@ def to_string_data(cls, data, depth):
...
+                val = unicode(val, 'utf-8')
+            except UnicodeDecodeError:
+                is_bin = True
+            if not is_bin:
+                s += '"'
+                # Escape backslash and double quotes.
+                s += val.replace('\\', '\\\\').replace('"', '\\"')
+                s += '"'
+            else:
+                s += '!!binary '
```
Comment (elhimov) [inline tools/tarantool-gdb.py:414]:
```suggestion
                s += '!!binarystr '
```

Comment (locker) [reply tools/tarantool-gdb.py]: Added `bin:` and `str:` to binary strings and dropped `!!binary` tag, as discussed.

**Review by elhimov — APPROVED** [review]

**Pull request merged**